ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -65,39 +67,45 @@ def __init__(self):
         self.waypoints = []
         self.pose = None
         self.velocity = None
-        self.prev_step_v = 0.0
-        self.prev_step_a = 0.0
+        self.stop_calc_v = 0.0  # measurement used to calculate stopping distance
+        self.stop_calc_a = 0.0  # measurement used to calculate stopping distance
+        self.stop_target = None
         self.lights = None
         self.final_waypoints = []
+        self.waypoints_2d = []  # store x,y pairs for use in KDTree
+        self.waypoint_tree = None  # store KDTree in here
         self.final_waypoints_start_ptr = 0
         self.back_search = False
         self.last_search_distance = None
         self.last_search_time = None
         self.dyn_test_stoplight = False
         self.next_tl_wp = -1  # None
-        self.dyn_tl_buffer = 5.0  # tunable distance to stop before tl wp
+        self.dyn_tl_buffer = 2.5  # tunable distance to stop before tl wp
+        self.dyn_creep_zone = 7.5  # should only creep forward in this buffer
         self.dyn_jmt_time_factor = 1.0  # tunable factor to make nicer s curve
         self.update_rate = 10
         self.max_velocity = 0.0  # set based on max velocity in waypoints
-        self.default_velocity = 10.7
-        self.lookahead_wps = 100  # 200 is too many
+        self.default_velocity = 0.0 #10.7
+        self.lookahead_wps = 20  # 200 is too many
         self.subs = {}
         self.pubs = {}
         self.dyn_reconf_srv = None
         self.max_s = 0.0  # length of track
         self.JMT_List = []
         # target max acceleration/braking force - dynamically adjustable
-        self.default_accel = 1.5
+        self.default_accel = 1.2
         self.state = 'stopped'  # for now only use to see if stopped or moving
         self.is_decelerating = True
         self.min_stop_distance = 0.0
         self.stopping_distance = 0.0
+        self.decel_at_min_moving_velocity = -0.5  # still decelerating
```

Comment:
Can we target `self.decel_at_min_moving_velocity = 0.0`?  I tried testing this and it seemed to make good profiles with a soft ending without dips.